service/matching/matching_engine_test.go

```diff
@@ -3025,6 +3460,14 @@ func (m *testTaskManager) CreateTasks(
 	taskType := request.TaskQueueInfo.Data.TaskType
 	rangeID := request.TaskQueueInfo.RangeID
 
+	// randomly returning an error
+	if m.generateErrorRandomly() {
+		return nil, &persistence.ConditionFailedError{
```

Comment:
should we also simulate errors where the update is performed but it returns a timeout error? (not in this PR, can be later)

Reply:
Why timeout specifically? How will this help?

Reply:
it doesn't have to be timeout, but in general, simulate the situation where we get an error but the write happens anyway. that's one of the trickier situations to handle so it might be worth it. I didn't have any specific test in mind though.